source/languages/en/riak/dev/taste-of-riak/object-modeling-nodejs.md

```diff
@@ -0,0 +1,115 @@
+timelines, so we can simply add that type into the key name. We will
+also want to partition each collection object into some time period,
+that way the object doesn't grow too large (see note below).
+
+For `Timelines`, let's use the pattern `<username>_<type>_<date>` for
+users, and `<groupname>_Inbox_<date>` for groups, which will look like
+`joeuser_SENT_2014-03-06` or `marketing_group_INBOX_2014-03-05`,
+respectively.
+
+<div class="note">
+<div class="title">Note</div>
+Riak performs best with objects under 1-2MB. Objects larger than that
+can hurt performance, especially many siblings are being created. We
+will cover siblings, sibling resolution, and sibling explosions in the
+next chapter.  </div>
+
+#### Keeping our story straight with repositories
+
+Now that we've figured out our object model, let's write some
+repositories to help create and work with these objects in Riak:
+
+* [Base `Repository` class](https://github.com/basho/taste-of-riak/blob/master/nodejs/Ch03-Msgy-Schema/repositories/repository.js)
+* [`UserRepository` class](https://github.com/basho/taste-of-riak/blob/master/nodejs/Ch03-Msgy-Schema/repositories/user-repository.js)
+* [`MsgRepository` class](https://github.com/basho/taste-of-riak/blob/master/nodejs/Ch03-Msgy-Schema/repositories/msg-repository.js)
+* [`TimelineRepository` class](https://github.com/basho/taste-of-riak/blob/master/nodejs/Ch03-Msgy-Schema/repositories/timeline-repository.js)
+* [`TimelineManager` class that manages `Msg` and `Timeline` objects](https://github.com/basho/taste-of-riak/blob/master/nodejs/Ch03-Msgy-Schema/timeline-manager.js)
+
+Finally, let's test them:
+
+[*Example:* Putting it all together](https://github.com/basho/taste-of-riak/blob/master/nodejs/Ch03-Msgy-Schema/app.js)
+
+As you can see, the repository pattern helps us with a few things:
+
+ - It helps us to see if an object exists before creating a new one
+ - It keeps our buckets and key names consistent
+ - It provides us with a consistent interface to work with.
+
+While this set of repositories solves many of our problems, it is very
+minimal and doesn't cover all the edge cases. For instance, what happens
+if two different people try to create a user with the same username?
+
+We can also easily "compute" key names now, but how do we quickly look
+up the last 10 messages a user sent? Many of these answers will be
+application dependent. If your application shows the last 10 messages in
+reverse order, for example, you may want to store that set of data in
+another collection object to make lookup faster. There are drawbacks to
+every solution, but we recommend seeking out the key/value-based
+solution first, as it will likely be the quickest.
+
+So to recap, in this chapter we learned:
+
+* How to choose bucket names
+* How to choose natural keys based on how we want to partition our data
+
+In the next chapter, we will learn about siblings, how to deal with
```

Comment:
They all end with vague "next chapter" wording. I can remove that from all or leave that to another PR.